### PR TITLE
ARROW-6954: [Python] [CI] Add Python 3.8 to CI matrix

### PR DESCRIPTION
**Pull request by pitrou**
Also use a separate conda environment rather than installing everything
in the base conda environment, as conda can complain about conflicts.

### CODE REVIEW

**Comment by pitrou** [comment]
@ursabot crossbow submit test-conda-python-3.8 test-conda-python-3.8-pandas-latest test-conda-python-3.8-dask-master

**Comment by pitrou** [comment]
@kszucs

File: ci/docker/conda-python.dockerfile [hunk 1]
```diff
@@ -22,13 +22,15 @@ FROM ${org}/${arch}-conda-cpp:latest
...
-RUN conda install -q \
+RUN conda install -n testenv -q \
         --file arrow/ci/conda_env_python.yml \
         python=${python} \
-        tensorflow \
         nomkl && \
     conda clean --all
 
+COPY python/requirements.txt /arrow/python/
+RUN bash -c "source activate testenv && pip install -r /arrow/python/requirements.txt"
```
Comment (jorisvandenbossche) [inline ci/docker/conda-python.dockerfile]: Why is this needed? 
If there is something missing (six?), should that be added to `conda_env_python.yml` ?

Comment (pitrou) [reply ci/docker/conda-python.dockerfile]: It's needed for 2.7 backports such as [futures](https://pypi.org/project/futures/). They must only be installed on Python 2 builds, and I don't know how to express that in a conda YAML file.

**Comment by xhochy** [comment]
What is the reason that ORC and flight are deactivated here?

**Comment by pitrou** [comment]
* Flight: because it's not used by the R bindings
* ORC: because there was a weird link error in that build and, since ORC is not relevant for testing HDFS, I decided to disable it instead of spending more time diagnosing the issue

File: ci/docker/conda-python.dockerfile (hunk 1 above)
Comment (kszucs) [reply ci/docker/conda-python.dockerfile]: We should find a better way to install the dependencies for python 2.

**Comment by kszucs** [comment]
I'm not in favor of introducing a testenv Conda environment, it makes the scripts and the interactive debugging more complicated, like `docker-compose run conda-python bash` should open a shell with the base environment activated. With testenv we need to activate it manually. If we can change the default conda environment from base to testenv then it's fine from me.

**Comment by pitrou** [comment]
> I'm not in favor of introducing a testenv Conda environment

This is mandatory for this PR. Conda refuses to switch the base Python version to 3.8 (it produces conflicts). Also, separating the base environment (which contains the conda install) from work environments is recommended practice.

>  If we can change the default conda environment from base to testenv then it's fine from me.

I'm not sure how to do that. Even if we change the `bash` defaults, it won't affect commands executed without a shell.

**Comment by kszucs** [comment]
@pitrou let me experiment a bit with using a different default conda environment then

**Comment by pitrou** [comment]
> BTW my base environment uses python 3.8 by default.

I doubt it. Miniconda only comes with Python 3.7.
https://docs.conda.io/en/latest/miniconda.html

**Comment by kszucs** [comment]
@ursabot crossbow submit test-conda-python-3.8 test-conda-python-3.8-pandas-latest test-conda-python-3.8-dask-master test-conda-python-2.7-pandas-master test-conda-python-3.7-spark-master  test-conda-python-3.7-turbodbc-master

**Comment by kszucs** [comment]
@ursabot crossbow submit test-conda-python-3.8 test-conda-python-3.8-pandas-latest test-conda-python-3.8-dask-master test-conda-python-2.7-pandas-master test-conda-python-3.7-spark-master test-conda-python-3.7-turbodbc-master

**Review by kszucs — APPROVED** [review]
LGTM, thanks @pitrou!
